src/Orleans.Core.Abstractions/Transactions/TransactionAttribute.cs

```diff
@@ -20,8 +20,8 @@ public TransactionAttribute(TransactionOption requirement)
 
     public enum TransactionOption
     {
-        RequiresNew,
-        Required,
-        NotSupported
+        NotSupported, // Logic is not transactional.  If called within the context of a transaction, the context will not be passed ot the call.
```

Comment:
The comment makes it sound like it's fine to call a `NotSupported` method from within the context of a transaction (it just wont be a part of the transaction)... but I would have assumed from the name that this means that transactions are *explicitly not supported* and therefore calling such a method from within a transaction is an error because it will not behave transactionally (and therefore breaks ACID).

A method which explicitly does not support transactions silently taking part in a transaction worries me.

Maybe a more appropriate name for the described behavior would be `NotApplicable`, as in "the notion of a transaction simply does not apply to this method". Eg, perhaps the method simply returns the number 5 whenever called and there's nothing which the transactions subsystem needs to do to ensure that behaves as expected.

Reply:
We could rename this to `Suppress`, aligning it with .NET's `TransactionScopeOption` enum.

Reply:
I think " If called within the context of a transaction, the context will not be passed ot the call. " is implementation details which may not mean much to the user. How about 

When you set this attribute value, orleans transactions runtime ensures that the method call never participates in a transaction, regardless of the transactional status of its caller. 


This is borrowed from COM+ transaction concept : https://docs.microsoft.com/en-us/windows/desktop/cossdk/configuring-transactions . They have a similar set of tx attr with us.

Reply:
Let's discuss the attributes meanings and names in the today's transaction sync.

Reply:
> When you set this attribute value, orleans transactions runtime ensures that the method call never participates in a transaction, regardless of the transactional status of its caller.

`Never` is another possible value, used by EJB transactions for their equivalent to this enum and it means "do not start a transaction and throw if there is an existing transaction"

Yep, let's discuss in the meeting